mmdet3d/models/data_preprocessors/data_preprocessor.py

```diff
@@ -378,10 +382,50 @@ def voxelize(self, points: List[torch.Tensor]) -> Dict[str, torch.Tensor]:
                 coors.append(res_coors)
             voxels = torch.cat(points, dim=0)
             coors = torch.cat(coors, dim=0)
+        elif self.voxel_type == 'cylindrical':
+            voxels, coors = [], []
+            for i, (res, data_sample) in enumerate(zip(points, data_samples)):
+                rho = torch.sqrt(res[:, 0]**2 + res[:, 1]**2)
+                phi = torch.atan2(res[:, 1], res[:, 0]) / torch.pi * 180
+                polar_res = torch.stack((rho, phi, res[:, 2]), dim=-1)
+                # TODO: implement cylindrical voxelization in voxel_layer
+                min_bound = polar_res.new_tensor(
+                    self.voxel_layer.point_cloud_range[:3])
+                max_bound = polar_res.new_tensor(
+                    self.voxel_layer.point_cloud_range[3:])
+                polar_res = torch.clip(polar_res, min_bound, max_bound)
```

Comment:
It seems that `torch.clip` support `Tensor` value for min and max since PyTorch 1.9.0. Whether can we use another method to support PyTorch < 1.9.0.

Reply:
Have renamed `torch.clip` to alias `torch.clamp` so that support low version Pytorch.